content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,39 +35,219 @@ are covered here as well as the `System.Diagnostics` API.
 
-## Traces
+## Example app preparation {#example-app}
+
+This page uses a modified version of the example app from
+[Getting Started](/docs/languages/net/getting-started/) to help you learn
+about manual instrumentation.
+
+You don't have to use the example app: if you want to instrument your own app or
+library, follow the instructions here to adapt the process to your own code.
+
+### Dependencies {#example-app-dependencies}
+
+- [.NET SDK](https://dotnet.microsoft.com/download/dotnet) 6+
+
+### Create and launch an HTTP Server
+
+To highlight the difference between instrumenting a library and a standalone
+app, split out the dice rolling into a library file, which then will be
+imported as a dependency by the app file.
+
+Create the library file named `Dice.cs` and add the following code to it:
+
+```csharp
+/*Dice.cs*/
+
+public class Dice
+{
+    private int min;
+    private int max;
+
+    public Dice(int min, int max)
+    {
+        this.min = min;
+        this.max = max;
+    }
+
+    public List<int> rollTheDice(int rolls)
+    {
+        List<int> results = new List<int>();
+
+        for (int i = 0; i < rolls; i++)
+        {
+            results.Add(rollOnce());
+        }
+        
+        return results;
+    }
+
+    private int rollOnce()
+    {
+        return Random.Shared.Next(min, max + 1);
+    }
+}
 
-### Initializing tracing
+```
+
+Create the app file `DiceController.cs` and add the following code to it:
 
-There are two main ways to initialize [tracing](/docs/concepts/signals/traces/),
-depending on whether you're using a console app or something that's ASP.NET
-Core-based.
+```csharp
+/*DiceController.cs*/
+using Microsoft.AspNetCore.Mvc;
+using System.Net;
 
-#### Console app
 
-To start tracing in a console app, you need to create a tracer provider.
+public class DiceController : ControllerBase
+{
+    private ILogger<DiceController> logger;
+
+    public DiceController(ILogger<DiceController> logger)
+    {
+        this.logger = logger;
+    }
+
+    [HttpGet("/rolldice")]
+    public List<int> RollDice(string player, int? rolls)
+    {
+        if(!rolls.HasValue)
+        {
+            logger.LogError("Missing rolls parameter");
+            throw new HttpRequestException("Missing rolls parameter", null, HttpStatusCode.BadRequest);
+        }
+            
+        var result = new Dice(1, 6).rollTheDice(rolls.Value);
+
+        if (string.IsNullOrEmpty(player))
+        {
+            logger.LogInformation("Anonymous player is rolling the dice: {result}", result);
+        }
+        else
+        {
+            logger.LogInformation("{player} is rolling the dice: {result}", player, result);
+        }
+
+        return result;
+    }
+}
+
+```
+
+Replace the content of the Program.cs file with the following code:
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+
+var app = builder.Build();
+
+app.MapControllers();
+
+app.Run();
+```
+
```

Comment:
~~~suggestion

In the `Properties` subdirectory, replace the content of `launchSettings.json` with the following:

```json
{
  "$schema": "http://json.schemastore.org/launchsettings.json",
  "profiles": {
    "http": {
      "commandName": "Project",
      "dotnetRunMessages": true,
      "launchBrowser": true,
      "applicationUrl": "http://localhost:8080",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    }
  }
}
```

~~~